atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestV2Table.java

```diff
@@ -0,0 +1,273 @@
+package com.palantir.atlasdb.table.description.generated;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.table.generation.ColumnValues;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.common.base.BatchingVisitableView;
+import com.palantir.common.persist.Persistables;
+import java.lang.Iterable;
+import java.lang.Long;
+import java.lang.String;
+import java.lang.SuppressWarnings;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Generated;
+
+@Generated("com.palantir.atlasdb.table.description.render.TableRendererV2")
+@SuppressWarnings("all")
+public class SchemaApiTestV2Table {
+    private static final String rawTableName = "SchemaApiTest";
+
+    private final Transaction t;
+
+    private final TableReference tableRef;
+
+    private SchemaApiTestV2Table(Transaction t, Namespace namespace) {
+        this.tableRef = TableReference.create(namespace, rawTableName);
+        this.t = t;
+    }
+
+    public static SchemaApiTestV2Table of(Transaction t, Namespace namespace) {
+        return new SchemaApiTestV2Table(t, namespace);
+    }
+
+    public static String getRawTableName() {
+        return rawTableName;
+    }
+
+    public TableReference getTableRef() {
+        return tableRef;
+    }
+
+    public String getTableName() {
+        return tableRef.getQualifiedName();
+    }
+
+    public Namespace getNamespace() {
+        return tableRef.getNamespace();
+    }
+
+    /**
+     * Returns the value for column Column1 and specified row components. */
+    public Optional<Long> getColumn1(String component1) {
+        SchemaApiTestTable.SchemaApiTestRow row = SchemaApiTestTable.SchemaApiTestRow.of(component1);
+        byte[] bytes = row.persistToBytes();
+        ColumnSelection colSelection = 
+                ColumnSelection.create(Collections.singletonList(PtBytes.toCachedBytes("c")));
+        RowResult<byte[]> rowResult = t.getRows(tableRef, ImmutableSet.of(bytes), colSelection).get(bytes);
```

Comment:
interesting, does this `get()` actually work? typically maps use the `.equals` method to compare keys, and `byte[]#equals` will only return true only if the other `byte[]` is the same object

Reply:
(basically this seems to require that `t.getRows` returns a map where the keys are the same byte arrays that were passed in)

Reply:
This actually works since at runtime a comparator is passed [here](https://github.com/palantir/atlasdb/blob/develop/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/Cells.java#L112), and TreeMaps use the custom comparator to find out if two keys are equal. Have tried it out with an in-memory table to doublecheck :) Same call is made in both API v1 and v2.

Reply:
ah, cool, makes sense